tests/capture/test_capture_transforms.py

```diff
@@ -0,0 +1,674 @@
+            (
+                qml.probs(wires=0),  # wires
+                qml.capture.AbstractMeasurement(
+                    qml.measurements.ProbabilityMP._abstract_eval, n_wires=1, has_eigvals=False
+                ),
+            ),
+            (
+                qml.sample(op=qml.measure(0)),  # MCM
+                qml.capture.AbstractMeasurement(
+                    qml.measurements.SampleMP._abstract_eval, n_wires=1
+                ),
+            ),
+        ],
+    )
+    def test_aval(self, val, expected_aval):
+        """Test that the abstract evaluation of a TransformTracer is set correctly."""
+        _, _, trace = init_test_variables()
+        tracer = TransformTracer(trace, val, 0)
+        assert tracer.aval == expected_aval
+
+    def test_full_lower(self):
+        """Test that TransformTracer.full_lower returns the same class."""
+        _, _, trace = init_test_variables()
+        tracer = TransformTracer(trace, 0, 0)
+        assert tracer.full_lower() is tracer
+
+    def test_repr(self):
+        """Test that the repr(TransformTracer) is correct."""
+        level = 0
+        sublevel = 1
+        val = 2
+        idx = 3
+
+        dummy_program, main, _ = init_test_variables(level=level)
+        trace = TransformTrace(main, sublevel, dummy_program)
+        tracer = TransformTracer(trace, val, idx)
+
+        expected_repr = (
+            f"TransformTracer(TransformTrace(level={level}/{sublevel}), val={val}, idx={idx})"
+        )
+        assert repr(tracer) == expected_repr
+
+
+@transform
+def non_plxpr_transform(tape):
+    """Dummy transform that does not provide a way to transform PLxPR."""
+    return [tape], lambda results: results[0]
+
+
+def _change_rotations_plxpr_transform(
+    primitive, tracers, params, targs, tkwargs, state
+):  # pylint: disable=unused-argument
+    """Convert RX to RY, RY to RZ, and RZ to RX PLxPR transform."""
+    # Step 1: Transform primitive
+    prim_map = {
+        "RX": qml.RY._primitive,
+        "RY": qml.RZ._primitive,
+        "RZ": qml.RX._primitive,
+    }
+    primitive = prim_map.get(primitive.name, primitive)
+    # Step 2: Update tracers
+    tracers = [
+        TransformTracer(t._trace, t.val, t.idx + 1) if isinstance(t, TransformTracer) else t
+        for t in tracers
+    ]
+    # Step 3: Return the result of the transformation
+    return primitive.bind(*tracers, **params)
+
+
+@partial(transform, plxpr_transform=_change_rotations_plxpr_transform)
+def change_rotations(tape):
+    """Convert RX to RY, RY to RZ, and RZ to RX tape transform."""
+    op_map = {
+        "RX": qml.RY,
+        "RY": qml.RZ,
+        "RZ": qml.RX,
+    }
+    new_ops = [
+        op_map[op.name](op.data[0], op.wires) if op.name in op_map else op for op in tape.operations
+    ]
+    new_tape = qml.tape.QuantumScript(
+        new_ops, tape.measurements, shots=tape.shots, trainable_params=tape.trainable_params
+    )
+    return [new_tape], lambda results: results[0]
```

Comment:
Is this code ever called (same for the `expval_to_var` transform)?

Reply:
No, it is not. But for the sake of correctness, I actually wanted to define the tape transform correctly. I'll add a comment to clarify this.